servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcUtils.java

```diff
@@ -71,10 +86,31 @@
     private static final CharSequence GRPC_USER_AGENT = newAsciiString("grpc-service-talk/");
     private static final CharSequence GRPC_MESSAGE_ENCODING_KEY = newAsciiString("grpc-encoding");
     private static final CharSequence GRPC_ACCEPT_ENCODING_KEY = newAsciiString("grpc-accept-encoding");
+    private static final CharSequence GRPC_TIMEOUT_KEY = newAsciiString("grpc-timeout");
     private static final GrpcStatus STATUS_OK = GrpcStatus.fromCodeValue(GrpcStatusCode.OK.value());
     private static final ConcurrentMap<List<ContentCodec>, CharSequence> ENCODINGS_HEADER_CACHE =
             new ConcurrentHashMap<>();
     private static final CharSequence CONTENT_ENCODING_SEPARATOR = ", ";
+    /**
+     * Unlike {@link #CONTENT_ENCODING_SEPARATOR} it is possible or (depending on usage) likely that we will encounter a
+     * very large number of different duration values so we need to implement an LRU cache to limit the number of
+     * cached header values.
+     */
+    private static final Map<Duration, CharSequence> TIMEOUT_HEADER_VALUES_CACHE =
+            Collections.synchronizedMap(new LinkedHashMap<Duration, CharSequence>() {
```

Comment:
What if we remove the cache now and consider it as a follow-up internal improvement? We should not overcomplicate for cases when the value is computed because the probability it will hit the cache is very low.

Reply:
I will remove the cache for now.